garglk/sndsdl.c

```diff
@@ -516,7 +647,21 @@ static glui32 play_mod(schanid_t chan, long len)
     FILE *file;
     char *tn;
     char *tempdir;
-    int music_busy;
+    int music_busy, loop;
+
+    if (chan == NULL)
+        gli_strict_warning("MOD player called with an invalid channel!");
```

Comment:
Since `chan` is dereferenced below, I presume the function should return here instead of continuing.